ksqldb-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlTarget.java

```diff
@@ -195,11 +199,16 @@ private KsqlRequest createKsqlRequest(
 
   private void executeRequestAsync(
       final String path,
-      final Object jsonEntity
+      final Object jsonEntity,
+      final boolean expectFailures
   ) {
     execute(HttpMethod.POST, path, jsonEntity, (resp, vcf) -> {
     }).exceptionally(t -> {
-      log.error("Unexpected exception in async request", t);
+      if (expectFailures) {
```

Comment:
This precedes even this change, but I feel we should throw an exception instead of returning `null`? and let the caller decide whether to log or not.. I am wondering if we should pass a flag down into a common class like `KsqlTarget`

Reply:
On second thought, guess the class itself is aware of different endpoints.. more about this method.. can it throw the exception and let the caller decide

Reply:
I'd agree with Vinoth here - methods like `postAsyncHeartbeatRequest` and `postAsyncLagReportingRequest` should return a `CompletableFuture`, which allows the caller to decide if the care about the response and if they do, what to do with it.

Reply:
I'd agree with Vinoth here - methods like `postAsyncHeartbeatRequest` and `postAsyncLagReportingRequest` should return a `CompletableFuture`, which allows the caller to decide if the care about the response and if they do, what to do with it.

Reply:
I was originally just keeping with an internal implementation that might be reusable for "fire and forget" type methods, but I'm also fine with making this a bit more general and letting callers handle the response/exception.

Changed to return `CompletableFuture<RestResponse<HeartbeatResponse>>` which can now be handled by the caller, including handling the exceptional case.